src/physics/em/detail/BhabhaEnergyDistribution.i.hh

```diff
@@ -41,8 +41,13 @@ CELER_FUNCTION real_type BhabhaEnergyDistribution::operator()(Engine& rng)
     const real_type g_denominator = this->calc_g_fraction(
         min_energy_fraction_, this->max_energy_fraction());
 
-    UniformRealDistribution<> sample_inverse_epsilon(
-        1 / this->max_energy_fraction(), 1 / min_energy_fraction_);
+    real_type inv_energy_min
+        = min(1 / this->max_energy_fraction(), 1 / min_energy_fraction_);
+    real_type inv_energy_max
+        = max(1 / this->max_energy_fraction(), 1 / min_energy_fraction_);
+
+    UniformRealDistribution<> sample_inverse_epsilon(inv_energy_min,
+                                                     inv_energy_max);
```

Comment:
Ah, this is what you meant by the "limits issue". I think rather than changing your code to this rather awkward construction, it's better to remove the check from `UniformRealDistribution`, which (although in accordance with the requirements of the equivalent `std::` distribution) doesn't in practice need that assertion at all.